File: Src/Web/src/app/products/models/product.ts
export class Product {
    public id: number;
    public name: string;
    public description: string;
    public price: number;
    public expiryDate: Date;
    public quantity: number;
    public deliveryOptions: string;
    public availability: string;
    constructor(name: string, price: number, expiryDate: Date, quantity: number, deliveryOptions: string, availability: string, id?: number, description?: string) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.expiryDate = expiryDate;
        this.quantity = quantity;
        this.deliveryOptions = deliveryOptions;
        this.availability = availability;
    }
}

File: Src/Microservices/Product-Service/ProductService.DomainService/Medifii.ProductService.Repositories/Models/Product.cs
using System;

namespace Medifii.ProductService.Repositories.Models
{
    public class Product
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public float Price { get; set; }

        public int Quantity { get; set; }

        public DateTime ExpiryDate { get; set; }

        public bool Availability { get; set; }
    }
}
